source/common/router/router.cc

```diff
@@ -694,6 +704,36 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
       *this, std::move(generic_conn_pool), can_send_early_data, can_use_http3);
   LinkedList::moveIntoList(std::move(upstream_request), upstream_requests_);
   upstream_requests_.front()->acceptHeadersFromRouter(end_stream);
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.streaming_shadow")) {
+    // start the shadow streams.
+    for (const auto& shadow_policy_wrapper : active_shadow_policies_) {
```

Comment:
Seems like we clobber `shadow_headers_` if we have multiple policies to just be based on a single shadow header from one policy?

https://github.com/envoyproxy/envoy/pull/24365/files#diff-e3584a650478334f7373c0df344c85d499b5123f5237ce2bbabaad5ee53b899eR683

Reply:
You have to make one copy per shadow, otherwise you get at least 2 problems: 
1. Ownership issues (shadow stream may outlive the main stream)
2. Upstream filters for one shadow may make header modifications that are undesired for other shadows.